tests/postgres/alltypes_test.go

```diff
@@ -931,6 +934,81 @@ func TestTimeExpression(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestIntervalSetFunctionality(t *testing.T) {
+	updateQuery := `
+UPDATE test_sample.employee
+SET pto_accrual = INTERVAL '3 HOUR'
+WHERE employee.employee_id = $1
+RETURNING employee.employee_id AS "employee.employee_id",
+          employee.first_name AS "employee.first_name",
+          employee.last_name AS "employee.last_name",
+          employee.employment_date AS "employee.employment_date",
+          employee.manager_id AS "employee.manager_id",
+          employee.pto_accrual AS "employee.pto_accrual";
+`
+	insertQuery := `
+INSERT INTO test_sample.employee (employee_id, first_name, last_name, employment_date, manager_id, pto_accrual)
+VALUES ($1, $2, $3, $4, $5, $6)
+ON CONFLICT (employee_id) DO UPDATE
+       SET pto_accrual = excluded.pto_accrual
+RETURNING employee.employee_id AS "employee.employee_id",
+          employee.first_name AS "employee.first_name",
+          employee.last_name AS "employee.last_name",
+          employee.employment_date AS "employee.employment_date",
+          employee.manager_id AS "employee.manager_id",
+          employee.pto_accrual AS "employee.pto_accrual";
+`
+
+	testCases := []struct {
+		expectedQuery    string
+		name             string
+		duration         string
+		expectedInterval string
+		statement        func(employee *model.Employee) jet.Statement
+	}{
+		{
+			name:             "updateQuery",
+			expectedQuery:    updateQuery,
+			duration:         "3h",
+			expectedInterval: "03:00:00",
+			statement: func(employee *model.Employee) jet.Statement {
+				return Employee.UPDATE(Employee.PtoAccrual).SET(
+					Employee.PtoAccrual.SET(INTERVAL(3, HOUR)),
+				).WHERE(Employee.EmployeeID.EQ(Int(1))).RETURNING(Employee.AllColumns)
+			},
+		},
+		{
+			expectedQuery:    insertQuery,
+			name:             "insertQuery",
+			duration:         "5h",
+			expectedInterval: "05:00:00",
+			statement: func(employee *model.Employee) jet.Statement {
+				return Employee.INSERT(Employee.AllColumns).
+					MODEL(employee).
+					ON_CONFLICT(Employee.EmployeeID).
+					DO_UPDATE(SET(
+						Employee.PtoAccrual.SET(Employee.EXCLUDED.PtoAccrual),
+					)).RETURNING(Employee.AllColumns)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		slog.Info("Running test", slog.Any("test", tc.name))
+		testutils.ExecuteInTxAndRollback(t, db, func(tx *sql.Tx) {
+			var windy model.Employee
+			windy.PtoAccrual = ptr.Of(tc.duration)
+			stmt := tc.statement(&windy)
+
+			testutils.AssertStatementSql(t, stmt, tc.expectedQuery)
+			err := stmt.Query(tx, &windy)
+			assert.Nil(t, err)
+			assert.Equal(t, *windy.PtoAccrual, tc.expectedInterval)
+
+		})
+	}
+}
```

Comment:
I'm aware that table-driven tests tend to be endorsed by the golang community, but I'm mostly against them since the drawbacks far exceed the benefits. Especially for integration tests. Some of the drawbacks are:
- unreadable. I've seen tests with 1000 test cases, and I've ended up scrolling up and down, just to figure out the test logic. Table-driven tests usually take 3x,4x, or more lines of code than regular tests.
- complicates debugging. I can't just put a breakpoint at a specific test case, I have to run them all(I know some IDEs provide help in this regard).
- tests are rigid. All test cases can do only one thing. In this case, execute the statement and check the result. For instance, if I want to check a slightly different scenario that affects only let's say `insertQuery` I can't, I have to write a different query.

For simple unit tests that check just one function, the rigid point does not apply, but unreadability and complicated debugging are the reasons I tend to avoid them completely even for unit tests. 

In this case, two explicit sub tests would suffice:

```golang
func TestIntervalSetFunctionality(t *testing.T) {

t.Run("update", 
t.Run("upsert",

}
```

Reply:
Oh, okay.  I misunderstood your request then.  Let me fix this up.